core/src/main/kotlin/br/com/concrete/yosef/api/DynamicViewCreator.kt

```diff
@@ -57,14 +60,45 @@ class DynamicViewCreator(
                     "not a ViewGroup.")
             }
 
-            it.forEach { addChildrenRecursively(parentView, it, listener) }
+            if (parentView is ListView) {
+                addChildrenToParentListView(it, parentView, context, listener)
+            } else {
+                it.forEach {
+                    addChildrenRecursively(parentView, it, listener)
+                }
+            }
         }
 
         component.applyProperties(parentView, componentSpec.dynamicProperties, listener)
 
         return parentView
     }
 
+    private fun addChildrenToParentListView(children: List<DynamicComponent>, listView: ListView, context: Context, listener: OnActionListener?) {
+        val listOfViews: ArrayList<View> = ArrayList()
+        children.forEach {
+            val comp = getComponentByType(it)
+            val view = comp.createView(listView.context)
```

Comment:
**nope nope nope** the whole point of not using a simple linear layout for this is that list view can create views as we need them. instead of creating them all at once, we can simply send the list of components down to the list view adapter and when it needs to create a view for a specific position, we query the list of components at that position and create its view.